File: backend/schemas/ticket.py
from pydantic import BaseModel
from typing import Optional
from uuid import UUID
from datetime import datetime


class TicketCreate(BaseModel):
    title: str
    description: str
    priority: Optional[str] = "low"
    category: Optional[str] = None


class TicketUpdate(BaseModel):
    status: Optional[str]
    priority: Optional[str]
    assigned_agent_id: Optional[UUID]


class TicketResponse(BaseModel):
    id: UUID
    title: str
    status: str
    priority: str
    category: Optional[str]
    sentiment: Optional[str]
    ai_confidence: Optional[int]
    is_auto_resolved: bool
    created_at: datetime

    class Config:
        orm_mode = True


class TicketMessageCreate(BaseModel):
    message: str

File: frontend/types/ticket.ts
export interface Ticket {
  id: string;
  tenant_id: string;
  title: string;
  description: string;
  status: "open" | "in_progress" | "closed";
  priority: "low" | "medium" | "high" | "urgent";
  created_at: string;
  updated_at: string;
  customer_id?: string;
  assigned_to?: string;
}

export interface TicketMessage {
  id: string;
  ticket_id: string;
  sender_type: string;
  sender_id: string;
  message: string;
  created_at: string;
}

export interface TicketCreate {
  title: string;
  description: string;
  priority: "low" | "medium" | "high" | "urgent";
  customer_id?: string;
}

export interface TicketUpdate {
  title?: string;
  description?: string;
  status?: "open" | "in_progress" | "closed";
  priority?: "low" | "medium" | "high" | "urgent";
  assigned_to?: string;
}

export interface TicketMessageCreate {
  message: string;
}
